appsec/src/helper/sampler.hpp

```diff
@@ -29,48 +27,19 @@ class sampler {
         }
         // NOLINTEND(cppcoreguidelines-avoid-magic-numbers,readability-magic-numbers)
     }
-    class scope {
-    public:
-        explicit scope(std::atomic<bool> &concurrent) : concurrent_(&concurrent)
-        {
-            concurrent_->store(true, std::memory_order_relaxed);
-        }
-
-        scope(const scope &) = delete;
-        scope &operator=(const scope &) = delete;
-        scope(scope &&oth) noexcept
-        {
-            concurrent_ = oth.concurrent_;
-            oth.concurrent_ = nullptr;
-        }
-        scope &operator=(scope &&oth)
-        {
-            concurrent_ = oth.concurrent_;
-            oth.concurrent_ = nullptr;
-
-            return *this;
-        }
 
-        ~scope()
-        {
-            if (concurrent_ != nullptr) {
-                concurrent_->store(false, std::memory_order_relaxed);
-            }
+    bool get()
+    {
+        bool result = false;
+        if (sample_rate_ == 1) {
+            return true;
         }
 
-    protected:
-        std::atomic<bool> *concurrent_;
-    };
-
-    std::optional<scope> get()
-    {
         const std::lock_guard<std::mutex> lock_guard(mtx_);
 
-        std::optional<scope> result = std::nullopt;
-
-        if (!concurrent_ && floor(request_ * sample_rate_) !=
-                                floor((request_ + 1) * sample_rate_)) {
-            result = {scope{concurrent_}};
+        if (floor(request_ * sample_rate_) !=
+            floor((request_ + 1) * sample_rate_)) {
+            result = true;
```

Comment:
All this code (including the lines below) can now be rewritten as (note that `request_ + 1` in your current version can overflow):

```cpp
auto old_request = request_;
request_ = (request_ + 1) < std::numeric_limits<unsigned>::max() ? request_ + 1 : 1;
return floor(old_request * sample_rate_) != floor((old_request + 1) * sample_rate_);